### ISSUE TITLE
Allow oversampled and/or under-sampled input to flow_from_dataframe

### ISSUE THREAD

**Issue by ResidentMario** [keras-team/keras-preprocessing#181]
I have an image dataset with an unbalanced class distribution: certain common classes have up to 10x as many samples as certain uncommon classes. I would like to rebalance what images my classifier is exposed to by using some combination of oversampling and undersampling methods from `imbalanced-learn`.

I used the following code to see if this was something I could achieve using `flow_from_dataframe`:

```python
X_meta = pd.read_csv("X_meta")  # get source metadata
import numpy as np

split_ratio = 0.8
n_samples = len(X)
split_idx = int(split_ratio * n_samples)

np.random.seed(42)
idxs = np.arange(len(X))
np.random.shuffle(idxs)

X_train, X_test = X[idxs[:split_idx]], X[idxs[split_idx:]]
y_train, y_test = y[idxs[:split_idx]], y[idxs[split_idx:]]

X = X_meta[['CroppedImageURL']].values
y = X_meta['LabelName'].values

from imblearn.over_sampling import RandomOverSampler
X, y = RandomOverSampler(random_state=42).fit_resample(X, y)

import keras
from keras.preprocessing.image import ImageDataGenerator

train_datagen = ImageDataGenerator(
    rotation_range=40,
    width_shift_range=0.2,
    height_shift_range=0.2,
    rescale=1/255,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True,
    fill_mode='nearest'
)

test_datagen = ImageDataGenerator(
    rescale=1/255
)

X_train_df = pd.DataFrame().assign(ImagePath=X_train[:, 0], ImageClass=y_train)
X_test_df = pd.DataFrame().assign(ImagePath=X_test[:, 0], ImageClass=y_test)

train_generator = train_datagen.flow_from_dataframe(
    X_train_df,
    directory='/Users/alex/Desktop/comet/data/images_cropped/',
    x_col='ImagePath',
    y_col='ImageClass',
    target_size=(48, 48),
    batch_size=512,
    class_mode='categorical'
)

validation_generator = test_datagen.flow_from_dataframe(
    X_test_df,
    directory='/Users/alex/Desktop/comet/data/images_cropped/',
    x_col='ImagePath',
    y_col='ImageClass',    
    target_size=(48, 48),
    batch_size=512,
    class_mode='categorical'
)
```

However, this didn't work&mdash;`flow_from_dataframe` deduplicates paths which appeared multiple times in its input `DataFrame`.

While this might not be something worth breaking backwards compatibility over in terms of default behavior, it'd be great if I had a switch to turn this deduplication behavior off. As far as I can tell this would be the easiest way to incorporate data resampling technique into my model training workflow; the workaround would be to manually copy the images on disk, which is super dissatisfying.

If that sounds like a good idea, maybe I can whip something up. 😄

**Comment by Vijayabhaskar96** [comment]
> However, this didn't work—flow_from_dataframe deduplicates paths which appeared multiple times in its input DataFrame.

Have you tried with `drop_duplicates=False` ?

**Comment by rragundez** [comment]
Like @Vijayabhaskar96 use `drop_duplicates=False`. Also i recommend you not to over or under sample but to use class weights to influence the loss function. 

`model.fit_generator(img_generator, class_weight=class_weight)`

**Comment by ResidentMario** [comment]
It's already a parameter! Now I feel very foolish indeed. Thank you.
